### PR TITLE
Feature/vote abstain

### PR DESCRIPTION
**Pull request by svojsu**
Add the ability to vote abstein

### CODE REVIEW

File: novawallet/Modules/Vote/Governance/ReferendumVoteSetup/ReferendumVoteSetupPresenter.swift [hunk 1]
```diff
@@ -308,15 +313,24 @@ final class ReferendumVoteSetupPresenter {
             factory: dataValidatingFactory,
             params: params,
             selectedLocale: selectedLocale,
+            convictionUpdateClosure: { [weak self] in
```
Comment (ERussel) [inline novawallet/Modules/Vote/Governance/ReferendumVoteSetup/ReferendumVoteSetupPresenter.swift:316]: 6 parameters to much. I would make a separate model to provide handlers: fee and conviction update

File: novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift [hunk 2]
```diff
@@ -53,6 +53,12 @@ protocol GovernanceValidatorFactoryProtocol: BaseDataValidatingFactoryProtocol {
         delegateId: AccountId?,
         locale: Locale?
     ) -> DataValidating
+
+    func dontNeedConvictionUpdate(
```
Comment (ERussel) [inline novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift:57]: Probably, ```voteMatchesConviction``` sounds better

**Review by ERussel — COMMENTED** [review]
We also need to ensure that we don't allow to vote abstain in Gov1. I would add supportsAbstainVoting calculating variable to shared state that calculates it based on ```settings.option == .governancev2```. Then pass it to modules

File: novawallet/Modules/Vote/Governance/ReferendumVoteSetup/ReferendumVoteSetupPresenter.swift [hunk 3]
```diff
@@ -308,15 +313,24 @@ final class ReferendumVoteSetupPresenter {
...
+                self?.selectConvictionValue(0)
+                self?.provideConviction()
+
+                self?.performValidation(
```
[the remark this answers is not in the document]
Comment (ERussel) [reply novawallet/Modules/Vote/Governance/ReferendumVoteSetup/ReferendumVoteSetupPresenter.swift:326]: The validation is synchronous. It means that update closure will be called before success closure, we just need to move validation forward when a user hits continue

File: novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift [hunk 4]
```diff
@@ -300,6 +307,34 @@ extension GovernanceValidatorFactory: GovernanceValidatorFactoryProtocol {
...
         })
     }
+
+    func voteMatchesConviction(
+        with newVote: ReferendumNewVote?,
+        selectedConviction: ConvictionVoting.Conviction?,
+        convictionUpdateClosure: @escaping () -> Void,
+        locale: Locale?
+    ) -> DataValidating {
+        ErrorConditionViolation(onError: { [weak self] in
```
Comment (ERussel) [inline novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift:317]: Now, I got it why we are rerunning the validation. The validation must be WarningConditionalViolation and a user can continue not aborting the runner

Comment (svojsu) [reply novawallet/Modules/Vote/Governance/Validating/GovernanceValidatorFactory.swift:317]: Oh, now I see. Yes, makes sense.

**Review by ERussel — APPROVED** [review]

**Pull request merged**